model_evaluation/Predict bias.ipynb

```diff
@@ -0,0 +1,370 @@
+    "direction_male = kernel_gender_male - np.sum(np.multiply(kernel_gender_male, kernel_gender_mean))* kernel_gender_mean\n",
+    "direction_female = kernel_gender_female - np.sum(np.multiply(kernel_gender_female, kernel_gender_mean))* kernel_gender_mean"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 107,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[-1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 107,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "cosine_similarity(direction_female, direction_male)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 114,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Compute mean\n",
+    "kernel_occupation_mean = np.mean(kernel_occupation, axis=0)\n",
+    "# Apply  x - np.sum(np.multiply(x, mean))* mean"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 119,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "accountant:-0.166758477688\n",
+      "acupuncturist:0.0150433778763\n",
+      "architect:-0.106728702784\n",
+      "attorney:-0.0355984941125\n",
+      "chiropractor:-0.112065583467\n",
+      "comedian:-0.17996160686\n",
+      "composer:-0.154989466071\n",
+      "dentist:-0.00389941781759\n",
+      "dietitian:0.00302037596703\n",
+      "dj:-0.156128510833\n",
+      "filmmaker:-0.116180986166\n",
+      "interior_designer:-0.00478803366423\n",
+      "journalist:-0.0217301938683\n",
+      "landscape_architect:-0.00763043016195\n",
+      "magician:-0.00733107328415\n",
+      "massage_therapist:-0.0116159021854\n",
+      "model:0.0549785941839\n",
+      "nurse:0.099561393261\n",
+      "painter:0.0174702480435\n",
+      "paralegal:0.0106164813042\n",
+      "pastor:-0.161623597145\n",
+      "personal_trainer:-0.133440434933\n",
+      "photographer:-0.0985902026296\n",
+      "physician:-0.00131351128221\n",
+      "poet:-0.061441861093\n",
+      "professor:0.00782079994678\n",
+      "psychologist:0.00208866596222\n",
+      "rapper:-0.112389668822\n",
+      "real_estate_broker:-0.000683411955833\n",
+      "software_engineer:-0.0237298682332\n",
+      "surgeon:-0.0968104675412\n",
+      "teacher:-0.0625882595778\n",
+      "yoga_teacher:0.0292760580778\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(33):\n",
+    "#     _bias = np.abs(cosine_similarity(kernel_gender_female, kernel_occupation[i].reshape(1, -1))) + \\\n",
+    "#                   np.abs(cosine_similarity(kernel_gender_male, kernel_occupation[i].reshape(1, -1)))\n",
+    "\n",
+    "    _bias = (cosine_similarity(kernel_gender_female - kernel_gender_male, kernel_occupation[i].reshape(1, -1)))\n",
+    "\n",
+    "#     _bias = cosine_similarity(direction_male, kernel_occupation[i].reshape(1, -1))\n",
```

Comment:
Ah, I see what you mean when you say there are multiple ways to compute this cosine similarity. I'll try a few of them but the direction_male, direction_female intuitively feel like good places to start.